src/main/java/com/google/javascript/gents/ModuleConversionPass.java

```diff
@@ -347,4 +421,85 @@ void registerLocalSymbol(String sourceFile, String fullLocalName, String require
     typeRewrite.put(sourceFile, fullLocalName, localName);
     typeRewrite.put(sourceFile, requiredNamespace, localName);
   }
+
+  /** Metadata about an exported symbol. */
+  private static class ExportedSymbol {
+    /** The name of the file exporting the symbol. */
+    final String filename;
+
+    /**
+     * The name that the symbol is declared in the module.
+     *
+     * <p>For example, {@code Foo} in: <code> class Foo {}</code>
+     */
+    final String localName;
+
+    /**
+     * The name that the symbol is exported under via named exports.
+     *
+     * <p>For example, {@code Bar} in: <code> exports {Bar}</code>.
+     *
+     * <p>If a symbol is directly exported, as in the case of <code>export class Foo {}</code>, the
+     * {@link #localName} and {@link #exportedName} will both be {@code Foo}.
+     */
+    final String exportedName;
+
+    private ExportedSymbol(String filename, String localName, String exportedName) {
+      this.filename = checkNotNull(filename);
+      this.localName = checkNotNull(localName);
+      this.exportedName = checkNotNull(exportedName);
+    }
+
+    static ExportedSymbol of(String filename, String localName, String exportedName) {
+      return new ExportedSymbol(filename, localName, exportedName);
+    }
+
+    static ExportedSymbol fromExportAssignment(
+        Node rhs, String exportedNamespace, String exportedSymbol, String filename) {
+      String localName = (rhs.getQualifiedName() != null) ? rhs.getQualifiedName() : exportedSymbol;
+
+      String exportedName;
+      if (exportedNamespace.equals(EXPORTS)) { // is a default export
+        exportedName = localName;
+      } else if (exportedNamespace.startsWith(EXPORTS)) { // is a named export
+        exportedName =
+            exportedNamespace.substring(EXPORTS.length() + 1, exportedNamespace.length());
+      } else { // exported via goog.provide
+        exportedName = "";
```

Comment:
I think I'd actually prefer to use `null` for this. Empty strings mostly behave like strings, so this might cause more confusion down the line.